test/framework/resource/k8s/jobs/manager.go

```diff
@@ -39,6 +43,44 @@ type defaultManager struct {
 	k8sClient client.Client
 }
 
+func (m *defaultManager) CreateJobAndWaitForJobToRun(ctx context.Context,
+	jobs *batchV1.Job) (*batchV1.Job, error) {
+	err := m.k8sClient.Create(ctx, jobs)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait till the cache is refreshed
+	time.Sleep(utils.PollIntervalShort)
+
+	observedJob := &batchV1.Job{}
+	return observedJob, wait.PollImmediateUntil(utils.PollIntervalShort, func() (bool, error) {
+		podList := &v1.PodList{}
+		err = m.k8sClient.List(ctx, podList, &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(jobs.Spec.Template.Labels),
+		})
+		if err != nil {
+			return false, err
+		}
+		if err := m.k8sClient.Get(ctx, utils.NamespacedName(jobs), observedJob); err != nil {
+			return false, err
+		}
+		if observedJob.Status.Succeeded > 0 || observedJob.Status.Failed > 0 {
+			return false, fmt.Errorf("some jobs has either"+
+				" succeeded or failed :%v", observedJob.Status)
+		} else if observedJob.Status.Active > 0 {
+			var count int
+			for _, pod := range podList.Items {
+				if pod.Status.Phase == v1.PodRunning && strings.Contains(pod.Name, jobs.Name) {
+					count++
+				}
+			}
+			return count == int(*jobs.Spec.Parallelism), nil
```

Comment:
will apiserver automatically fills this Parallelism? wondering whether it's always non-nil

Reply:
I tried creating a Job without specifying parallelism and I could see in the spec that the parallelism was auto populated.
```
spec:
  backoffLimit: 4
  completions: 1
  parallelism: 1
```